Don't select items on mouse right-click

TxHawks opened the pull request:
I think it is a bit counter intuitive that right-clicking an item fires the select function.

I suggest only firing the select function if the `mousedown` event is a right click (`evt.which === 1`).

TxHawks committed:
Don't select items on mouse right-click

LeaVerou commented on awesomplete.js:
```diff
@@ -95,7 +95,7 @@ var _ = function (input, o) {
 			}
 
-			if (li) {
+			if (li && evt.which === 1) {  // Only select on left click
```
Given that `evt.which` is non-standard, I'd rather do `evt.which !== 2 && evt.which !== 3` or `[2,3].indexOf(evt.which) === -1` or something like that.

vlazar replied:
Or https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button ?

TxHawks replied:
My bad. 

I guess we should probably go with the standard and less convoluted `evt.button === 0` which [seem to be well supported](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button).

LeaVerou commented:
Thanks! See comment.

TxHawks committed:
Only select on mouse left-click, test w\ e.button

Check which mouse button was clicked using the standard `event.button`
rather than the nonstandard `event.which`.

TxHawks committed:
Add changes to minified file

LeaVerou commented:
Yes! Nice find!
Merged, thanks!
